### ISSUE TITLE
leafletProxy for tmap

### ISSUE THREAD

**Issue by mtennekes** [mtennekes/tmap#284]
A couple of times I have created a shiny dashboard using tmap, but the annoying thing is that after the input has been changed, the map resets its focus and zoom level. For leaflet, there is `leafletProxy` to deal with this, which can probably be also used for tmap. I'll have to figure out how, but tips are welcome.

**Comment by tim-salabim** [comment]
Same here for mapview ;-)

**Comment by mtennekes** [comment]
Done! See the minimal example https://github.com/mtennekes/tmap/blob/master/examples/tmapOutput.R

So far, I've only implemented `tm_remove_layer`, which removes any polygon, symbol, line or raster layer, and the corresponding legend controls. I've also reorganized the panes. Every tmap layer (expect for basemaps) will be assigned to a pane `tmapXXX`, where `XXX corresponds to the z-index, which is by default 401, 402, etc. Users are able to specify custom zindex values.

I would appreciated if someone could test it (@Nowosad @Robinlovelace ), and check whether the docs are clear. I plan to submit a new version to CRAN soon.
